dmoj/problem_schema.yml

```diff
@@ -0,0 +1,142 @@
+#normal names
+type: map
+mapping:
+  "archive":
+    type: str
+    pattern: .+\.zip
+  "generator":
+    type: str
+  "checker":
+    type: any
+  "signature_grader":
+    type: map
+    mapping:
+      "entry":
+        type: str
+      "header":
+        type: str
+  "custom_judge":
+    type: str
+  "io_redirects":
+    type: map
+    mapping:
+      "DATA.in":
+        type: map
+        mapping:
+          "fd":
+            type: str
+          "mode":
+            type: str
+      "DATA.out":
+        type: map
+        mapping:
+          "fd":
+            type: str
+          "mode":
+            type: str
+  "hints":
+    type: seq
+    sequence:
+      - type: str
+
+#defaults
+  "wall_time_factor":
+    type: int
+  "output_prefix_length":
+    type: int
+  "output_limit_length":
+    type: int
+
+#names with ++
```

Comment:
inb4 `output_prefix_length+: __import__('random').randint(1, 100)` in some April Fools problem, though whoever does that really should be punished severely.